packages/kubernetes-1.25/kubelet-exec-start-conf

```diff
@@ -0,0 +1,31 @@
+[Service]
+ExecStart=
+ExecStart=/usr/bin/kubelet \
+{{#unless settings.kubernetes.standalone-mode}}
+    --cloud-provider {{default "external" settings.kubernetes.cloud-provider}} \
```

Comment:
Sorry, this is actually the place we need to change. We can just delete this line and fall back to `--cloud-provider=external` for all variants since in-tree cloud provider code is being completely removed from `kubelet`.

This would also mean `settings.kubernetes.cloud-provider` is going to be an inert setting that doesn't do anything, but that's probably fine? @bcressey

Reply:
Or we can choose to change the default setting here:
https://github.com/bottlerocket-os/bottlerocket/blob/b5b6540cc96891548744c9e2041b12c03922fa86/sources/models/shared-defaults/kubernetes-aws.toml#L6

to "external" so we get to keep the setting until the time when kubelet actually removes the `--cloud-provider` flag.

Reply:
I think we need to change it to be explicit in the `kubelet-exec-start-conf` file. All k8s variants use the `shared-defaults` settings, so if we changed that to `external`, the earlier versions of k8s that should still be using in-tree providers would pick that up.

Reply:
> All k8s variants use the shared-defaults settings

Ah, I almost forgot about that. You're absolutely right.